src/browser/ui/dom/CSSPropertyOperations.js

```diff
@@ -64,6 +68,33 @@ if (__DEV__) {
       name.charAt(0).toUpperCase() + name.slice(1) + '?'
     );
   };
+
+  var warnStyleValueWithSemicolon = function(name, value) {
+    if (warnedStyleValues.hasOwnProperty(value) && warnedStyleValues[value]) {
+      return;
+    }
+    
+    warnedStyleValues[value] = true;
+    warning(
+      false,
+      'Style property values shouldn\'t contain a semicolon. Try "' +
+      name + ': ' + value.replace(badStyleValueWithSemicolonPattern, '') + '" instead.'
+    );
+  };
+
+  /**
+   * @param {string} name
+   * @param {string|number} value
+   */
+  var assertValidStyle = function(name, value) {
```

Comment:
I think it might be better to call it something with `warn` in the name. `assert` has the connotation that an error will be thrown.

Also, I think you'll probably want `@param {*} value` (unless we're doing a validation before this). There's nothing actually stopping you from saying `style={{foo: [1,2,3]}}`